extension/builtin/default_registry.go

```diff
@@ -36,62 +29,20 @@ var (
 	MySQLURIFlag = flag.String("mysql_uri", "test:zaphod@tcp(127.0.0.1:3306)/test", "uri to use with mysql storage")
 )
 
-// Default implementation of extension.Registry.
-type defaultRegistry struct {
-	db *sql.DB
+// UseSQLDB configures the registry to use the given SQL database for all storage.
+func UseSQLDB(r *extension.Registry, db *sql.DB) {
```

Comment:
+1 for `New*`.

In fact, I'd be tempted to either delete UseSQLDB entirely or move it to a testonly package. I don't see it getting used for non-test code.